src/main/scala/firrtl/passes/InferTypes.scala

```diff
@@ -103,8 +103,9 @@ object CInferTypes extends Pass {
         types(sx.name) = sx.tpe
         sx
       case sx: DefNode =>
-        types(sx.name) = sx.value.tpe
-        sx
+        val sxx = (sx map infer_types_e(types)).asInstanceOf[DefNode]
+        types(sxx.name) = sxx.value.tpe
+        sxx
```

Comment:
Why does `DefRegister` call `infer_types_e` on it's child `Expressions` but `DefWire` does not?